diff --git a/timelib/timelib_structs.h b/timelib/timelib_structs.h
index 5033321..5185e7b 100644
--- a/timelib/timelib_structs.h
+++ b/timelib/timelib_structs.h
@@ -23,7 +23,37 @@
 
 #include "timelib_config.h"
 
-#include "php_stdint.h"
+#ifdef HAVE_SYS_TYPES_H
+#include <sys/types.h>
+#endif
+
+#if defined(HAVE_INTTYPES_H)
+#include <inttypes.h>
+#elif defined(HAVE_STDINT_H)
+#include <stdint.h>
+#endif
+
+#ifdef PHP_WIN32
+/* TODO: Remove these hacks/defs once we have the int definitions in main/ 
+	 rathen than in each 2nd extension and win32/ */
+# include "win32/php_stdint.h"
+#else
+# ifndef HAVE_INT32_T
+#  if SIZEOF_INT == 4
+typedef int int32_t;
+#  elif SIZEOF_LONG == 4
+typedef long int int32_t;
+#  endif
+# endif
+
+# ifndef HAVE_UINT32_T
+#  if SIZEOF_INT == 4
+typedef unsigned int uint32_t;
+#  elif SIZEOF_LONG == 4
+typedef unsigned long int uint32_t;
+#  endif
+# endif
+#endif
 
 #include <stdio.h>
 
